src/test/java/core/basesyntax/services/impl/ShopReportServiceImplTest.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.services.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.exceptions.NullDataException;
+import core.basesyntax.services.ShopReportService;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ShopReportServiceImplTest {
+    private static ShopReportService reportService;
+    private static String valueSeparator;
+    private static String titleRow;
+    private static Map<String, Integer> processedData;
+
+    @BeforeClass
+    public static void beforeClass() {
+        valueSeparator = ",";
+        titleRow = "fruit,quantity";
+        processedData = Map.of("apple", 20, "banana", 100, "watermelon", 4);
+    }
+
+    @Before
+    public void setUp() {
+        reportService = new ShopReportServiceImpl(valueSeparator, titleRow);
```

Comment:
This is strange that you have such values passed in a constructor instead of constants, let's change it. 
Same for other similar cases

Reply:
++